compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -41,6 +44,33 @@ impl MetaVarExpr {
         check_trailing_token(&mut tts, sess)?;
         let mut iter = args.trees();
         let rslt = match ident.as_str() {
+            "concat" => {
```

Comment:
Can you add a separate feature gate for this with a test? Don't want this to accidentally wind up stabilized if https://github.com/rust-lang/rust/pull/122808 lands first.

A separate tracking issue will be good too, to discuss this syntax specifically.